### PR TITLE
Add a FileProviderExtension to Teacher

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: none
affects: none
release note: none

### CODE REVIEW

**Comment by vanwagonet** [comment]
Is there anything in here we should merge in regardless of finishing the file provider?
Is this something that is worthwhile to complete?

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/xo6dfie) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fxo6dfie)
teacher: [Link](https://rebrand.ly/zxch4sm) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fzxch4sm)
parent: [Link](https://rebrand.ly/g7jifre) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fg7jifre)

File: Core/Core/Files/GetFolders.swift [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        var folders: [APIFolder] = []
+        var resp: URLResponse?
+        var err: Error?
+        let group = DispatchGroup()
+        let handleCourses = { (courses: [APICourse]?, response: URLResponse?, error: Error?) in
+            resp = response
+            err = error
+            courses?.forEach { course in
+                group.enter()
+                environment.api.makeRequest(GetFolderRequest(context: .course(course.id.value), id: "root")) { (folder, _, error) in
```
Comment (inst-danger) [inline Core/Core/Files/GetFolders.swift:79]:
```suggestion
                environment.api.makeRequest(GetFolderRequest(context: .course(course.id.value), id: "root")) { (folder, _, _) in
```

File: Core/Core/Files/GetFolders.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            }
+            group.leave()
+        }
+        if let url = pageURL {
+            group.enter()
+            environment.api.exhaust(GetNextRequest<[APICourse]>(path: url.absoluteString), callback: handleCourses)
+        } else {
+            if let userID = environment.currentSession?.userID { // "self" doesn't work
+                group.enter()
+                environment.api.makeRequest(GetFolderRequest(context: .user(userID), id: "root")) { (folder, _, error) in
```
Comment (inst-danger) [inline Core/Core/Files/GetFolders.swift:95]:
```suggestion
                environment.api.makeRequest(GetFolderRequest(context: .user(userID), id: "root")) { (folder, _, _) in
```

File: rn/Teacher/ios/TeacherFileProvider/FileProviderExtension.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        } catch {
+            completionHandler(error)
+        }
+    }
+
+    override func startProvidingItem(at url: URL, completionHandler: @escaping (Error?) -> Void) {
+        guard let id = persistentIdentifierForItem(at: url), let remoteURL = fileItem(for: id)?.url else {
+            return completionHandler(NSFileProviderError(.noSuchItem))
+        }
+        env.api.makeDownloadRequest(remoteURL) { (tmpURL, response, error) in
```
Comment (inst-danger) [inline rn/Teacher/ios/TeacherFileProvider/FileProviderExtension.swift:71]:
```suggestion
        env.api.makeDownloadRequest(remoteURL) { (tmpURL, _, error) in
```

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/Files/GetFolders.swift:79:124: Unused Closure Parameter Violation: Unused parameter "error" in a closure should be replaced with _. (unused_closure_parameter)
    /Users/vagrant/git/Core/Core/Files/GetFolders.swift:95:113: warning: Unused Closure Parameter Violation: Unused parameter "error" in a closure should be replaced with _. (unused_closure_parameter)
    [student]
    ⚠️  /Users/vagrant/git/rn/Teacher/ios/TeacherFileProvider/FileProviderExtension.swift:95:1: Line Length Violation: Line should be 200 characters or less: currently 271 characters (line_length)
    /Users/vagrant/git/rn/Teacher/ios/TeacherFileProvider/FileProviderExtension.swift:71:59: warning: Unused Closure Parameter Violation: Unused parameter "response" in a closure should be replaced with _. (unused_closure_parameter)
    [parent]
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**